temperature/services.py

```diff
@@ -0,0 +1,35 @@
+import asyncio
+import os
+
+import httpx
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from city.crud import get_all_cities
+from temperature import schemas, crud
+
+API_URL = "http://api.weatherapi.com/v1"
+ENDPOINT = "/current.json"
+client = httpx.AsyncClient()
+
+
+async def update_temperature(db: AsyncSession):
+    cities = await get_all_cities(db)
+
+    async def fetch_temperature(city):
+        res = await client.get(
+            f"{API_URL}{ENDPOINT}?key={os.getenv('WEATHER_API_KEY')}&q={city.name}"
```

Comment:
Ensure that the `WEATHER_API_KEY` environment variable is set correctly in your environment to avoid authentication issues with the weather API.